### PR TITLE
[TBCM-305] Duplicate message when adding user

### PR DESCRIPTION
**Pull request by JGH-EY**

### CODE REVIEW

File: apps/api/src/user/user.controller.ts [hunk 1]
```diff
@@ -10,14 +10,15 @@ import {
...
+  UnprocessableEntityException,
 } from '@nestjs/common';
 import { UserService } from './user.service';
 import { ApiTags } from '@nestjs/swagger';
 import { CreateUserInviteDTO, EditUserDTO, PaginationRO, Role, UserRO } from '@tbcm/common';
 import { IRequest } from 'src/common/app-request';
 import { AllowRoles } from 'src/auth/allow-roles.decorator';
 import { FindUsersDto } from './dto/find-users.dto';
-
+import { QueryFailedError } from 'typeorm';
```
Comment (dbayly-freshworks) [inline apps/api/src/user/user.controller.ts:21]: Nit, best have a new line between imports and the start of code.

File: apps/web/src/services/useHttp.ts [hunk 2]
```diff
@@ -17,6 +17,8 @@ export const useHttp = () => {
       toast.error(toastMessage);
     } else if (err?.response?.status === 400) {
       toast.error('Kindly verify the input');
+    } else if (err?.response?.status === 422) {
+      toast.error(err.response.data.errorMessage);
```
Comment (dbayly-freshworks) [inline apps/web/src/services/useHttp.ts:21]: Nit - Might be worth changing this to a switch statement.

**Review by dbayly-freshworks — APPROVED** [review]
2 nits, otherwise looks fine to me

File: apps/api/src/user/user.controller.ts [hunk 3]
```diff
@@ -27,9 +28,20 @@ export class UserController {
...
+    try {
+      const tokenUser = req.user;
+      const user = await this.userService.createUserFromInvite(data, tokenUser);
+      return new UserRO(user);
+    } catch (err) {
+      if (
+        err instanceof QueryFailedError &&
+        err.message.includes('duplicate key value violates unique constraint')
+      ) {
+        throw new UnprocessableEntityException('User already exists');
```
Comment (kushal-arora-fw) [inline apps/api/src/user/user.controller.ts:40]: I'd suggest using ConflictException when the user already exists. StatusCode - 409

**Review by kushal-arora-fw — APPROVED** [review]
Looks good!

**Pull request merged**